Distributions/UnitTests/Interpolation/wlTestInterpolation.F90

```diff
@@ -0,0 +1,215 @@
+ 
+  linear_value = x1 + 2.0d0 * x2 - 1.0d0/3.0d0 * axis3_lo_res(1) - 7.0d0 * axis4_lo_res(1) + axis5_lo_res(1)
+  write(stdout,*) 'interp linear data 2D 2D absolute error', interp_result_linear-linear_value
+
+  !test 3D 3D linear data
+  interp_result_linear = LOG10(LinearInterp_Array_Point(ix1,ix2,ix3, &
+                                                        dx1,dx2,dx3,0.0d0,linear_data(:,:,:,1,1)))
+ 
+  linear_value = x1 + 2.0d0 * x2 - 1.0d0/3.0d0 * x3 - 7.0d0 * axis4_lo_res(1) + axis5_lo_res(1)
+  write(stdout,*) 'interp linear data 3D 3D absolute error', interp_result_linear-linear_value
+
+  !test 4D 4D linear data
+  interp_result_linear = LOG10(LinearInterp_Array_Point(ix1,ix2,ix3,ix4, &
+                                                        dx1,dx2,dx3,dx4,0.0d0,linear_data(:,:,:,:,1)))
+
+  linear_value = x1 + 2.0d0 * x2 - 1.0d0/3.0d0 * x3 - 7.0d0 * x4 + axis5_lo_res(1)
+  write(stdout,*) 'interp linear data 4D 4D absolute error', interp_result_linear-linear_value
+
+  !test 5D 5D linear data
+  interp_result_linear = LOG10(LinearInterp_Array_Point( ix1, ix2, ix3, ix4, ix5, &
+                                                         dx1, dx2, dx3, dx4, dx5, &
+                                                         0.0d0, linear_data))
+
+  linear_value = x1 + 2.0d0 * x2 - 1.0d0/3.0d0 * x3 - 7.0d0 * x4 + x5
+  write(stdout,*) 'interp linear data 5D 5D absolute error', interp_result_linear-linear_value
+
+
+  write(stdout,*) 'Testing interpolation with nonlinear source data and two different source array resolutions (factor of 2)'
+  write(stdout,*) 'source(x1,x2,x3,x4,x5) = sin(x1) * cos(x2) * sqrt(x3) + cos(x4 * sin(x5))'
+
+  !test 5D 5D nonlinear data
+  interp_result_non_linear_lo = LOG10(LinearInterp_Array_Point( ix1, ix2, ix3, ix4, ix5, &
+                                                                dx1, dx2, dx3, dx4, dx5, &
+                                                                0.0d0, non_linear_data_lo_res))
+
+
+  CALL GetIndexAndDelta( x1, axis1_hi_res, ix1, dx1 )
+  CALL GetIndexAndDelta( x2, axis2_hi_res, ix2, dx2 )
+  CALL GetIndexAndDelta( x3, axis3_hi_res, ix3, dx3 )
+  CALL GetIndexAndDelta( x4, axis4_hi_res, ix4, dx4 )
+  CALL GetIndexAndDelta( x5, axis5_hi_res, ix5, dx5 )
+
+  interp_result_non_linear_hi = LOG10(LinearInterp_Array_Point( ix1, ix2, ix3, ix4, ix5, &
+                                                                dx1, dx2, dx3, dx4, dx5, &
+                                                                0.0d0, non_linear_data_hi_res))
+
+  non_linear_value = sin(x1) * cos(x2) * sqrt(x3) + cos(x4 * sin(x5)) 
+
+  write(stdout,*) 'interp nonlinear data 5D 5D relative error lo res', &
+                  (interp_result_non_linear_lo-non_linear_value)/non_linear_value
+
+  write(stdout,*) 'interp nonlinear data 5D 5D relative error hi res', &
+                  (interp_result_non_linear_hi-non_linear_value)/non_linear_value
+
+END PROGRAM wlTestInterpolation
```

Comment:
We need a simple way to signal that the test has passed or failed if we ever want to incorporate this into a C.I. suite.

This could just be writing "PASSED" or "FAILED" to `stdout`, but could also be writing to a new file, e.g. "test_status".